### DISCUSSION TITLE
STEP400 Rackmount

### DISCUSSION THREAD

**Discussion by 7sorkr** [ponoor/step-series-support#24] [Show and tell]
![image0](https://user-images.githubusercontent.com/32606306/186663735-59d4f1fa-5c07-4081-a309-85316815b6c2.jpeg)

https://user-images.githubusercontent.com/32606306/186663421-c6f35bc1-4e49-4118-b7b4-8c26935f3d6c.MOV

I made the STEP400 into a 2U Rack
It can be done in 1U or in half-rack.

Please keep that in mind~

This is the current front and back contents.

**[Front]**
- Micro SD
- Power button (to be added)

**[Back]**
- Power
- Ethernet
- USB (C Type TO A Type)
- Top (Sensor Home & Limit)
- Bottom (Motor 1,2,3,4)

I put two sensors in one Ethernet terminal
Could it be a problem because the power is turned on?

I'd like to add it to the Front and Back, is it possible?

*** Back**
1. DIP Switch

*** Front**
1. Basic playback, stop button, etc
2. Motor ID (7 segment)
3. Motor speed (7 segment)
4. Position (7 segment)
5. Power usage (7 segment or off-the-shelf)
6. A screen that can express all of the above

**Comment by kanta** [comment]
Wow this is so serious professional stuff, thank you for sharing this!
FYI, I sometimes use this kind of boxes to bring STEP400 to stages or studios.
![image](https://user-images.githubusercontent.com/1097902/186841026-663503ca-64c8-47ff-af0a-38e20b274edb.png)

I think there is no problem for using RJ45 connector for 2 sensors. 

If you have unused LIMIT switch inputs, you can use them for playback/stop button.
But using a screen or 7 segment LEDs are too complicated job for STEP400, because the MCU is busy for many other tasks. I recommend to add an extra board dedicated for a screen control. Then we need a communication from STEP400 to the new board. There are some options to do that;
- Ethernet
- I2C
- Serial

If I were to do the same thing, I would make the new board also have a LAN connection so that the information could be sent via the PC. That way STEP400 would not have to change anything.

**Reply by 7sorkr** [reply]
![40887840-D05D-4503-8D58-F0F561CC17F3](https://user-images.githubusercontent.com/32606306/186846674-032ecb44-77b2-4ecf-8b7a-01a1c236f588.jpeg)


![EB9D1C0C-0F7B-43D5-BA5E-FA4967041B38](https://user-images.githubusercontent.com/32606306/186846719-cb256d5a-c81e-4415-87b1-2b66ab3aa837.jpeg)


There are still a lot of beginner's skills, so please refer to it

It's not organized
Just keep that in mind
